Compiled from "CWE197_Numeric_Truncation_Error__int_random_to_byte_16.java"
public class testcases.CWE197_Numeric_Truncation_Error.s02.CWE197_Numeric_Truncation_Error__int_random_to_byte_16 extends testcasesupport.AbstractTestCase {
  public testcases.CWE197_Numeric_Truncation_Error.s02.CWE197_Numeric_Truncation_Error__int_random_to_byte_16();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method testcasesupport/AbstractTestCase."<init>":()V
       4: return

  public void bad() throws java.lang.Throwable;
    Code:
       0: new           #7                  // class java/security/SecureRandom
       3: dup
       4: invokespecial #9                  // Method java/security/SecureRandom."<init>":()V
       7: invokevirtual #10                 // Method java/security/SecureRandom.nextInt:()I
      10: istore_1
      11: iload_1
      12: i2b
      13: invokestatic  #14                 // Method testcasesupport/IO.writeLine:(B)V
      16: return

  private void goodG2B() throws java.lang.Throwable;
    Code:
       0: iconst_2
       1: istore_1
       2: iload_1
       3: i2b
       4: invokestatic  #14                 // Method testcasesupport/IO.writeLine:(B)V
       7: return

  public void good() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: invokevirtual #20                 // Method goodG2B:()V
       4: return

  public static void main(java.lang.String[]) throws java.lang.ClassNotFoundException, java.lang.InstantiationException, java.lang.IllegalAccessException;
    Code:
       0: aload_0
       1: invokestatic  #25                 // Method mainFromParent:([Ljava/lang/String;)V
       4: return
}
